jekyll/_cci2/api-developers-guide.md

```diff
@@ -0,0 +1,476 @@
+
+5. One of the benefits of the CircleCI API v2 is the ability to remotely trigger pipelines with parameters. The following code snippet simply triggers a pipeline via `curl` without any body parameters:
+
+    ```sh
+    curl -X POST https://circleci.com/api/v2/project/gh/{YOUR_USER_NAME}/hello-world/pipeline \
+      -H 'Content-Type: application/json' \
+      -H 'Accept: application/json' \
+      -H "Circle-Token: $CIRCLECI_TOKEN" \
+      
+    # Which returns:
+    {
+      "number": 2,
+      "state": "pending",
+      "id": "e411ea74-c64a-4d60-9292-115e782802ed",
+      "created_at": "2020-01-15T15:32:36.605Z"
+    }
+    ```
+
+While this alone can be useful, we want to be able to customize parameters of the pipeline when we send this POST request. By including a body parameter in the `curl` request (via the `-d` flag), we can customize specific attributes of the pipeline when it runs: pipeline parameters, the branch, or the git tag. Below, we are telling the pipelines to trigger for "my-branch"
+
+    ```sh
+    curl -X POST https://circleci.com/api/v2/project/gh/teesloane/hello-world/pipeline \
+      -H 'Content-Type: application/json' \
+      -H 'Accept: application/json' \
+      -H "Circle-Token: $CIRCLE_TOKEN" \
+      -d '{ "branch": "bar" }' 
+    ```
+
+6. Let's move on to a more complex example: triggering a pipeline and passing a parameter that can be dynamically substituted into your configuration. In this example, we will pass a docker image tag to our docker-executor key. First, we will need to modify the `.circleci/config.yml` to be a little more complex than the standard "Hello World" sample provided by the onboarding.
+
+    ```yaml
+    version: 2.1
+    parameters:
+      image-tag:
+        type: string
+        default: "latest"
+    
+    jobs:
+      build:
+        docker:
+          - image: circleci/node:<< pipeline.parameters.image-tag >>
+        environment:
+          IMAGETAG: << pipeline.parameters.image-tag >>
+        steps:
+          - run: echo "Image tag used was ${IMAGETAG}"
+    ```
+
+You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we definte an "image-tag" to be expected in the JSON payload of a POST request to the _Trigger New Pipeline_ endpoint.
+
+7. Now we can run a `curl` request that passes variables in a POST request, similar to the following:
+
+    ```sh
+    curl -u ${CIRCLECI_TOKEN}: -X POST --header "Content-Type: application/json" -d '{
+      "parameters": {
+        "image-tag": "4.8.2"
+      }
+    }' https://circleci.com/api/v2/project/gh/{YOUR_USER_NAME}/hello-world/pipeline
+    ```
+
+This concludes the end-to-end example of using the V2 API. For more detailed information about other endpoints you may wish to call, please refer to the [CircleCI API v2 Documentation]({{site.baseurl}}/api/v2/#circleci-api) for an overview of all endpoints currently available.
+
+# Example API Use Cases
```

Comment:
Arguably the section above was already an example use case.

Reply:
fixed